instaclustr/resource_cluster.go

```diff
@@ -601,8 +649,209 @@ func appendIfMissing(slice []string, toAppend string) []string {
+}
+
+func hasKafkaSizeChanges(bundleIndex int, d *schema.ResourceData) bool {
+	return len(getNewSizeOrEmpty(d, getBundleOptionKey(bundleIndex, "zookeeper_node_size"))) > 0 ||
+		len(getNewSizeOrEmpty(d, "node_size")) > 0
+}
+
+func hasCassandraSizeChanges(d *schema.ResourceData) bool {
+	return len(getNewSizeOrEmpty(d, "node_size")) > 0
+}
+
+func doClusterResize(client *APIClient, clusterID string, d *schema.ResourceData, bundles []Bundle) error {
+	cluster, err := client.ReadCluster(clusterID)
+	if err != nil {
+		return fmt.Errorf("[Error] Error reading cluster: %s", err)
+	}
+	bundleIndex, err := getBundleIndex(cluster.BundleType, bundles)
+	if err != nil {
+		return err
+	}
+	switch cluster.BundleType {
+	case "APACHE_CASSANDRA":
+		if hasCassandraSizeChanges(d) {
+			return doLegacyCassandraClusterResize(client, cluster, d)
+		} else {
+			return nil
+		}
+	case "ELASTICSEARCH":
+		if hasElasticsearchSizeChanges(bundleIndex, d) {
+			return doElasticsearchClusterResize(client, cluster, d, bundleIndex)
+		} else {
+			return nil
+		}
+	case "KAFKA":
+		if hasKafkaSizeChanges(bundleIndex, d) {
+			return doKafkaClusterResize(client, cluster, d, bundleIndex)
+		} else {
+			return nil
+		}
+	default:
+		return fmt.Errorf("CDC resize does not support: %s", cluster.BundleType)
+	}
+}
+
+func getNewSizeOrEmpty(d *schema.ResourceData, key string) string {
+	if !d.HasChange(key) {
+		return ""
+	}
+	_, after := d.GetChange(key)
+	return after.(string)
+}
+
+func isElasticsearchSizeAllChange(kibanaSize, masterSize, dataSize string, kibana, dataNodes bool) (string, bool) {
+	if len(masterSize) == 0 {
+		return "", false
+	}
+	if kibana && (len(kibanaSize) == 0 || kibanaSize != masterSize) {
+		return "", false
+	}
+	if dataNodes && (len(dataSize) == 0 || dataSize != masterSize) {
+		return "", false
+	}
+	return masterSize, true
+}
+
+func getSingleChangedElasticsearchSizeAndPurpose(kibanaSize, masterSize, dataSize string, kibana, dataNodes bool) (string, NodePurpose, error) {
+	changedCount := 0
+	var nodePurpose NodePurpose
+	var nodeSize string
+	if len(masterSize) > 0 {
+		changedCount += 1
+		nodePurpose = ELASTICSEARCH_MASTER
+		nodeSize = masterSize
+	}
+	if len(dataSize) > 0 {
+		if !dataNodes {
+			return "", "", fmt.Errorf("[ERROR] This cluster does not have data only nodes, so data_node_sise is not used for this cluster. Please use master_node_size to change the size instead")
+		}
+		changedCount += 1
+		nodePurpose = ELASTICSEARCH_DATA_AND_INGEST
+		nodeSize = dataSize
+	}
+	if len(kibanaSize) > 0 {
+		if !kibana {
+			return "", "", fmt.Errorf("[ERROR] This cluster didn't enable Kibana, kibana_node_sise is not used for this cluster. Please use master_node_size to change the size instead")
+		}
+		changedCount += 1
+		nodePurpose = ELASTICSEARCH_KIBANA
+		nodeSize = kibanaSize
+	}
+	if changedCount > 1 {
+		return "", "", fmt.Errorf("[ERROR] Please change one size at a time or change all to same size at onece")
```

Comment:
"[ERROR] Please change one size at a time or change all to same size at onece" 

would replace with:

"[ERROR] Please change either a single node size at a time or change all nodes to the same size at once"

Reply:
Fixed